tests/test_iterator.rs

```diff
@@ -266,6 +266,51 @@ fn test_seek_for_prev() {
 }
 
 #[test]
+fn test_prefix_same_as_start() {
+    let path = TempDir::new("_rust_rocksdb_prefix_same_as_start").expect("");
+    let mut bbto = BlockBasedOptions::new();
+    bbto.set_bloom_filter(10, false);
+    bbto.set_whole_key_filtering(false);
+    let mut cf_opts = ColumnFamilyOptions::new();
+    cf_opts.set_block_based_table_factory(&bbto);
+    cf_opts
+        .set_prefix_extractor(
+            "FixedPrefixTransform",
+            Box::new(FixedPrefixTransform { prefix_len: 2 }),
+        )
+        .unwrap();
+    cf_opts.set_memtable_prefix_bloom_size_ratio(0.1 as f64);
+    let mut opts = DBOptions::new();
+    opts.create_if_missing(true);
+    let db = DB::open_cf(
+        opts,
+        path.path().to_str().unwrap(),
+        vec![("default", cf_opts)],
+    ).unwrap();
+
+    let wopts = WriteOptions::new();
+
+    db.put_opt(b"k1-0", b"c", &wopts).unwrap();
+    db.put_opt(b"k2-1", b"a", &wopts).unwrap();
+    db.flush(true /* sync */).unwrap();
+
+    db.put_opt(b"k2-2", b"b", &wopts).unwrap();
+    db.put_opt(b"k2-4", b"c", &wopts).unwrap();
+
+    let mut readopts = ReadOptions::new();
+    readopts.set_prefix_same_as_start(true);
+    let mut iter = db.iter_opt(readopts);
+
+    iter.seek_for_prev(SeekKey::Key(b"k2-3"));
+    assert!(iter.valid());
+    assert_eq!(b"k2-2", iter.key());
+
+    iter.seek_for_prev(SeekKey::Key(b"k2-0"));
+    // assert!(!iter.valid());
+    assert_eq!(b"k1-0", iter.key());
```

Comment:
Here iterator should be invalid.

Reply:
I think so too, but i don't understand why, is it the problem of my test?

Reply:
Have you figured out why? Need some help?

Reply:
not yet, can you take a look at my test?

Reply:
thanks

Reply:
so add a comment here, and then go on.